api/v1alpha1/warehouse_types.go

```diff
@@ -216,6 +230,8 @@ type WarehouseStatus struct {
 	// ObservedGeneration represents the .metadata.generation that this Warehouse
 	// was reconciled against.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// FreightReference refers to the last Freight produced by this Warehouse
+	FreightReference string `json:"freightReference,omitempty"`
```

Comment:
@maksimstankevic there is a `FreightReference` type. I propose something like this:

```suggestion
	LastFreight *FreightReference `json:"lastFreight,omitempty"`
```

The benefits of this are twofold:

1. No need to make a separate call later to retrieve Freight.
2. Guards against the possibility that the last Freight shipped from the Warehouse has since been deleted.

Reply:
@krancour, thanks, absolutely, I'm completely new to this k8s programming so still need to wrap my mind around a lot of ways :) I will refactor that once testing is successful. But I have a couple of points to raise, during my work on this I saw 2 instances of what I believe are bugs, can you give your take on those:

1)   I think it should not be possible to create a Warehouse with duplicated Git type subscriptions, verification should be added I suppose, below is an excerpt from a Warehouse running in my DEV env:
```
  uid: 46e4a742-0a97-48d5-8f9a-a719922820cb
spec:
  subscriptions:
  - git:
      branch: helm
      commitSelectionStrategy: NewestFromBranch
      repoURL: https://github.com/maksimstankevic/kargo-demp-876.git
  - git:
      commitSelectionStrategy: NewestFromBranch
      repoURL: https://github.com/maksimstankevic/kargo-demp-877.git
  - git:
      commitSelectionStrategy: NewestFromBranch
      repoURL: https://github.com/maksimstankevic/kargo-demp-877.git
status:
  freightReference: 70c665c38cf6f238e7fb0a6b7763ea902b29547d  
```
2) the interesting one: per my testing I see that in case of multiple Git supscriptions in a Warehouse a valid failure to get a new commit from one of Git repo subscriptions blocks normal progress of others, meaning new valid commits in other Git subscriptions will not be able to produce new Freight if one Git subscription fails to select a commit due to valid reasons (scanPaths/ignorePaths or Tag configurations): https://github.com/akuity/kargo/blob/c9aebcd37e20d40f75c31504297126fafc926968/internal/controller/warehouses/git.go#L58C1-L60C29
Don't you think that in such case we should just let the failing Git repo subscription to stay where it was is but if others have new valid commits - we should produce new Freight?

Reply:
> I think it should not be possible to create a Warehouse with duplicated Git type subscriptions, verification should be added I suppose

Additional validations seldom hurt. I would suggest opening a new issue for it and not dealing with it here.

> per my testing I see that in case of multiple Git supscriptions in a Warehouse a valid failure to get a new commit from one of Git repo subscriptions blocks normal progress of others

This is by design. If you think of a piece of Freight as a box containing all the artifacts required for your application, including image, manifests, etc., a failure to fetch any one of those means you have an incomplete set.

> Don't you think that in such case we should just let the failing Git repo subscription to stay where it was

I guess this suggests that if I'm looking for versions of x and y and I find x, but don't find y, we should use the last known version of y... the problem is that if there were a version of y that fit your current constraints, _it would have been found_, meaning the last known version of y _does not_ fit your constraints. The main scenario where I can see this happening is when someone has _changed_ their constraints. They previously wanted ^1.0.0 of y and now want ^2.0.0. If nothing matching ^2.0.0 was found, it's incorrect to move forward with the old y that matched ^1.0.0.

Reply:
yes, that makes sense, for validation of duplications I'll open an issue, that was my thought initially

Reply:
@krancour I pushed the changes, please, review when you get a chance

Reply:
@maksimstankevic at a glance, this is looking like it's almost there. I promise a more thorough review tomorrow (Tuesday).

Let me say again how much we appreciate this. This was an ambitious feature to bite off and it's coming along nicely!